jenkins/scripts/select-compiler.sh

```diff
@@ -112,4 +113,23 @@ elif [ "$SELECT_ARCH" = "AIXPPC" ]; then
     # front of PATH
     echo "Compiler set to default at 4.8.5"
   fi
+
+  elif [ "$SELECT_ARCH" = "MACOS" ]; then
+
+  echo "Setting compiler for Node version $NODEJS_MAJOR_VERSION on macos"
+
+  if [ "$NODEJS_MAJOR_VERSION" -ge "13" ]; then
+    # set xcode version to latest
+    sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
+    echo "set xcode to latest - At least version 10"
+  elif [ "$NODEJS_MAJOR_VERSION" -ge "12" ]; then
+    sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
```

Comment:
How does this select a different version? ` /Applications/Xcode.app/Contents/Developer` looks identical to the section for 13?

Reply:
Atm it is identical becauase 12 and 13 both build on 10 +. The actual min. xcode doesnt change to version 10 until node 13 (according to BUILDING.md) so the question is do we want to use latest for these versions or hardcode in a stable version and only change it if required (security patches etc.). 

For example `/Applications/Xcode.11.1.app/Contents/Developer` for node 13 and then something like `/Applications/Xcode10.x.app/Contents/Developer` for node 12. 

Depends if we want as few xcodes on the machine as possible or if we want a stable xcode for each node release?

Reply:
stable for each release, decided at time of first `.0.0` for that release line, never changed afterwards.

Reply:
Are these documented anywhere as to which version is used for each release?

Reply:
in the BUILDING.md on the relevant release branch of nodejs/node, if documented, if not.... https://github.com/nodejs/build/issues/1522 :-(

Reply:
We're locked in to 8 for Node 12. We've historically locked in for the entirety of a release line. That could change and it would be lower risk on macOS than others, but not zero risk and the TSC would have to sign off on it and Build WG would shoulder the blame if things break. So .. for now consider it locked to Xcode 8 for Node 12.